ArduCopter/autoyaw.cpp

```diff
@@ -56,40 +56,28 @@ void Mode::AutoYaw::set_mode(autopilot_yaw_mode yaw_mode)
     if (_mode == yaw_mode) {
         return;
     }
+    _last_mode = _mode;
     _mode = yaw_mode;
 
     // perform initialisation
     switch (_mode) {
 
-    case AUTO_YAW_LOOK_AT_NEXT_WP:
-        // wpnav will initialise heading when wpnav's set_destination method is called
-        break;
-
```

Comment:
This diff looks a little weird here.  This is becuase I changed _mode to be of type autopilot_yaw_mode so needed to add the default case.  I then tidied up the cases and preserved the comments (below).  This has been done as a seperate commit to make it clearer.